### PR TITLE
feat: [PL-26653]: return latest commit id of the file in getFile SCM API response

### PR DESCRIPTION
**Pull request by mohitg0795**
## Describe your changes

Test cases:

bitbucket cloud

- create pipeline
- update pipeline via harness
- do random commits on random files on same branch on git
- update pipeline via harness
- do random commits on git
- open pipeline from listing page and then update pipeline on harness

bitbucket onprem (master branch)

- create pipeline
- update pipeline via harness
- do random commits on random files on same branch on git
- update pipeline via harness
- do random commits on git
- open pipeline from listing page and then update pipeline on harness

ADO

- create pipeline
- update pipeline via harness
- do random commits on random files on same branch on git
- update pipeline via harness
- do random commits on git
- open pipeline from listing page and then update pipeline on harness

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35651)

### CODE REVIEW

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmFacilitatorServiceImpl.java [hunk 1]
```diff
@@ -424,4 +432,28 @@ private List<UserRepoResponse> convertToUserRepo(List<Repository> allUserRepos)
...
     return isNotEmpty(response.getReposList()) && isNotEmpty(response.getRepos(0).getNamespace());
   }
+
+  private ApiResponseDTO getGetFileAPIResponse(
+      ScmConnector scmConnector, FileContent fileContent, GetLatestCommitOnFileResponse getLatestCommitOnFileResponse) {
+    if (ScmApiErrorHandlingHelper.isFailureResponse(fileContent.getStatus(), scmConnector.getConnectorType())) {
+      return ApiResponseDTO.builder().statusCode(fileContent.getStatus()).error(fileContent.getError()).build();
+    }
+    if (isNotEmpty(getLatestCommitOnFileResponse.getError())) {
+      return ApiResponseDTO.builder().statusCode(400).error(getLatestCommitOnFileResponse.getError()).build();
```
Comment (bhavya181) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmFacilitatorServiceImpl.java:442]: why is this error handling, we need to add error handlers for them

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmFacilitatorServiceImpl.java` line 442 at r2](https://reviewable.io/reviews/harness/harness-core/35651#-N7a3p1zba8owHN_Kq5E-r2-442:-N7l_YezDmtso6eKXOQw:b-tdqz1h) ([raw file](https://github.com/harness/harness-core/blob/1693469b0767934101f8e95b8c00a23ab98a1315/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmFacilitatorServiceImpl.java#L442)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

why is this error handling, we need to add error handlers for them
</blockquote></details>

Currently, the SCM API doesn't pass any status code, thus its a generic error handling for now.

**Review by bhavya181 — APPROVED** [review]
> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35651)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Review by bhavya181 — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35651)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 3 of 4 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35651)* status: :shipit: complete! ✓

**Pull request merged**